base_layer/wallet/src/output_manager_service/storage/sqlite_db/mod.rs

```diff
@@ -415,64 +416,98 @@ impl OutputManagerBackend for OutputManagerSqliteDatabase {
         let acquire_lock = start.elapsed();
-        let status = if confirmed {
-            OutputStatus::Unspent as i32
-        } else {
-            OutputStatus::UnspentMinedUnconfirmed as i32
-        };
+
         debug!(
             target: LOG_TARGET,
-            "`set_received_output_mined_height` status: {}", status
+            "`set_received_outputs_mined_height_and_statuses` for {} outputs",
+            updates.len()
         );
-        let hash = hash.to_vec();
-        let mined_in_block = mined_in_block.to_vec();
-        let timestamp = NaiveDateTime::from_timestamp_opt(mined_timestamp as i64, 0).ok_or(
-            OutputManagerStorageError::ConversionError {
-                reason: format!("Could not create timestamp mined_timestamp: {}", mined_timestamp),
-            },
-        )?;
-        diesel::update(outputs::table.filter(outputs::hash.eq(hash)))
-            .set((
-                outputs::mined_height.eq(mined_height as i64),
-                outputs::mined_in_block.eq(mined_in_block),
-                outputs::status.eq(status),
-                outputs::mined_timestamp.eq(timestamp),
-                outputs::marked_deleted_at_height.eq::<Option<i64>>(None),
-                outputs::marked_deleted_in_block.eq::<Option<Vec<u8>>>(None),
-                outputs::last_validation_timestamp.eq::<Option<NaiveDateTime>>(None),
-            ))
-            .execute(&mut conn)
-            .num_rows_affected_or_not_found(1)?;
+
+        let mut query = String::from(
+            "INSERT INTO outputs ( commitment, mined_height, mined_in_block, status, mined_timestamp, spending_key, \
+             value, output_type, maturity, hash, script, input_data, script_private_key, sender_offset_public_key, \
+             metadata_signature_ephemeral_commitment, metadata_signature_ephemeral_pubkey, metadata_signature_u_a, \
+             metadata_signature_u_x, metadata_signature_u_y, spending_priority, covenant, encrypted_data, \
+             minimum_value_promise
+            )
+             VALUES ",
+        );
+
+        for update in &updates {
+            query.push_str(&format!(
+                "(x'{}', {}, x'{}', {}, '{}', 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0), ",
+                update.commitment.to_hex(),
+                update.mined_height as i64,
+                update.mined_in_block.to_hex(),
+                if update.confirmed {
+                    OutputStatus::Unspent as i32
+                } else {
+                    OutputStatus::UnspentMinedUnconfirmed as i32
+                },
+                if let Some(val) = NaiveDateTime::from_timestamp_opt(update.mined_timestamp as i64, 0) {
+                    val.to_string()
+                } else {
+                    "NULL".to_string()
+                },
+            ));
+        }
+
+        // Remove the trailing comma
+        query.pop();
+        query.pop();
```

Comment:
```suggestion
        // Remove the trailing comma, and space
        query.pop();
        query.pop();
```

I feel like the need for a `pop,pop` is a code smell. The normal way to handle this is a loop that generates a `Vec<String>` that you `join(', ')`, and then finally `query.push()`